app/components/post_list/post_list.js

```diff
@@ -343,18 +348,49 @@ export default class PostList extends PureComponent {
         );
     };
 
+    onContentSizeChange = (contentWidth, contentHeight) => {
+        this.contentHeight = contentHeight;
+        this.props.onContentSizeChange(this.state.postListHeight, contentHeight);
+
+        if (this.state.postListHeight && contentHeight < this.state.postListHeight && this.props.extraData) {
+            // We still have less than 1 screen of posts loaded with more to get, so load more
+            this.props.onLoadMoreUp();
+        }
+    };
+
     onLayout = (event) => {
         const {height} = event.nativeEvent.layout;
-        this.setState({
-            postListHeight: height,
-        });
+        this.setState({postListHeight: height});
+    };
+
+    onScroll = (event) => {
+        const contentOffset = event.nativeEvent.contentOffset.y;
+        if (contentOffset > 0) {
+            this.pageOffsetY = contentOffset;
+            this.contentHeight = event.nativeEvent.contentSize.height;
+            const direction = (this.contentOffsetY < contentOffset) ?
+                ListTypes.VISIBILITY_SCROLL_UP :
+                ListTypes.VISIBILITY_SCROLL_DOWN;
+            this.contentOffsetY = contentOffset;
+
+            if (
+                direction === ListTypes.VISIBILITY_SCROLL_UP &&
+                (this.contentHeight - this.pageOffsetY) < (this.state.postListHeight * 3.5)
```

Comment:
Can you maybe make a couple constants for how far from the top/bottom we load? Also, why use different amounts for both?